src/mkdocs_gallery/binder.py

```diff
@@ -199,10 +199,19 @@ def _copy_binder_notebooks(gallery_conf, mkdocs_conf):
 def check_binder_conf(binder_conf):
     """Check to make sure that the Binder configuration is correct."""
 
-    # Grab the configuration and return None if it's not configured
-    binder_conf = {} if binder_conf is None else binder_conf
-    if not isinstance(binder_conf, dict):
+    if not isinstance(binder_conf, (dict, None)):
         raise ConfigError('`binder_conf` must be a dictionary or None.')
+
+    # Set empty dict if user did not provide any configuration values
+    if binder_conf is None:
+        binder_conf = {}
+    vals = binder_conf.values()
+    default_values = [None, 'gh-pages', 'https://mybinder.org']
+    if vals == []:
+        binder_conf = {}
+    elif all([val in default_values for val in vals]):
```

Comment:
This is not very satisfying. What if someone sets `'branch'` to `'https://mybinder.org'` ? Also it duplicates the information in `GalleryPlugin` . I would rather find a better way to do this with mkdocs... 
I'll try to have a look one of these days but as you could notice I have a hard time finding a few minutes to code :( If you can find your way through mkdocs config class to create a "fixed" `MySubConfig` it would probably better.